### PR TITLE
deprecation(semver): deprecate `eq()`, `neq()`, `lt()`, `lte()`, `gt()` and `gte()`

### PR DESCRIPTION
**Pull request by timreichen**
ref: https://github.com/denoland/deno_std/issues/3948#issuecomment-1873464725

- deprecates `eq()`, `neq()`, `lt()`, `lte()`, `gt()` and `gte()`
- adds common use cases to `compare()` jsdoc

### CODE REVIEW

File: semver/eq.ts [hunk 1]
```diff
@@ -2,8 +2,12 @@
 import { compare } from "./compare.ts";
 import type { SemVer } from "./types.ts";
 
-/** Returns `true` if they're logically equivalent, even if they're not the exact
- * same version object. */
+/**
+ * Returns `true` if they're logically equivalent, even if they're not the exact same version object.
+ * @deprecated (will be removed in 0.213.0) use {@linkcode compare} instead.
```
Comment (iuioiua) [inline semver/eq.ts:7]:
```suggestion
 * @deprecated (will be removed in 0.213.0) Use {@linkcode compare} instead.
```
Ditto for other instances.

File: semver/eq.ts [hunk 2]
```diff
@@ -2,8 +2,12 @@
...
+ *
+ * This is equal to `compare(s0, s1) === 0`
```
Comment (iuioiua) [inline semver/eq.ts:9]:
```suggestion
 * This is equal to `compare(s0, s1) === 0`.
```
Ditto for other instances.

File: semver/compare_test.ts [hunk 3]
```diff
@@ -54,3 +54,171 @@ Deno.test({
...
+        const s0 = parse(v0);
+        const s1 = parse(v1);
+
+        const eq = compare(s0, s1);
+        const op = expected ? "==" : "!=";
+        assertEquals(eq, expected, `${s0} ${op} ${s1}`);
+      });
+    }
+  },
+});
```
Comment (iuioiua) [inline semver/compare_test.ts:101]: This is the only test needed. It takes care of all of the other equality and inequality cases more precisely.

**Review by iuioiua — CHANGES_REQUESTED** [review]
I agree with the overall idea of this PR. I have just a few nits.

It'd be good to hear other's opinions on this change.

**Comment by kt3k** [comment]
I'd prefer to keep these methods as they are easier to remember than the meaning of return values of `compare`

**Comment by timreichen** [comment]
> I'd prefer to keep these methods as they are easier to remember than the meaning of return values of `compare`

Ok, I will close this in favour of another PR.

**Pull request closed without merging**